### DISCUSSION TITLE
What is the function of universe joint?

### DISCUSSION THREAD

**Discussion by liuyidesktop** [stack-of-tasks/pinocchio#2161] [General]
After I imported the urdf file, the njoints variable was always 1 more than the actual joint. I printed the name of the joint and found that there was a universe joint. I saw on issues that this joint was meaningless. What should I pay attention to in actual use? Which variables in the model object are related to the root njoints, and which ones are related to the actual number of joints?

**Comment by florent-lamiraux** [comment]
Hello @liuxiaoyiyixiaoliu .
The main purpose of the 'universe' joint is to hold geometric objects that are fixed to the world frame, like static obstacles.
Otherwise, you can safely ignore this additional joint.

**Reply by joao-pm-santos96** [reply]
Hello @florent-lamiraux , thank for the answer but if the obstacles are static, why is there the need for a joint? Is there or could there be an option to not "import" the universe joint? Thank you.

**Reply by florent-lamiraux** [reply]
I think this is a design choice made at the beginning of the project and that has remained in the successive versions of pinocchio because nobody questioned it until now. There is no option to remove the 'universe' joint. I do not think it would be a good practice to add an option for that. What does the existence of this joint prevent you to do ?

**Reply by joao-pm-santos96** [reply]
In practice, nothing, it's just my curiosity
